fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java

```diff
@@ -73,27 +86,89 @@ public static ColumnRangePredicate andRange(
                 }
             }
         }
-        return new ColumnRangePredicate(rangePredicate.columnRef, TreeRangeSet.create(ranges));
+        return new ColumnRangePredicate(rangePredicate.getExpression(), TreeRangeSet.create(ranges));
     }
 
     public static ColumnRangePredicate orRange(
             ColumnRangePredicate rangePredicate, ColumnRangePredicate otherRangePredicate) {
         TreeRangeSet<ConstantOperator> result = TreeRangeSet.create();
         result.addAll(rangePredicate.columnRanges);
         result.addAll(otherRangePredicate.columnRanges);
-        return new ColumnRangePredicate(rangePredicate.getColumnRef(), result);
+        return new ColumnRangePredicate(rangePredicate.getExpression(), result);
     }
 
     public boolean isUnbounded() {
         return columnRanges.asRanges().stream().allMatch(range -> !range.hasUpperBound() && !range.hasLowerBound());
     }
 
+    public List<ColumnRangePredicate> getEquivalentRangePredicates() {
+        if (isCastDate() || isStr2Date()) {
+            return getEquivalentRangePredicateForDate();
+        }
+        return Lists.newArrayList();
+    }
+
+    boolean isCastDate() {
+        if (!(expression instanceof CastOperator)) {
+            return false;
+        }
+        CastOperator castOperator = expression.cast();
+        return castOperator.getChild(0).isColumnRef()
+                && castOperator.getChild(0).getType().isStringType()
+                && castOperator.getType().isDate();
+    }
+
+    boolean isStr2Date() {
+        if (!(expression instanceof CallOperator)) {
+            return false;
+        }
+        CallOperator callOperator = expression.cast();
+        // check whether is str2date(columnref, '%Y-%m-%d')
+        return callOperator.getFnName().equalsIgnoreCase(FunctionSet.STR2DATE)
+                && callOperator.getChild(0).isColumnRef() &&
+                ((ConstantOperator) callOperator.getChild(1)).getChar().equals("%Y-%m-%d");
```

Comment:
should support all `%Y %m %d` format ?

Reply:
maybe we could only support "%Y-%m-%d" now, because date's to_string is this format